week3/homework/src/methodList.js

```diff
@@ -0,0 +1,150 @@
+const fs = require('fs');
+const uuid = require('uuid/v4');
+
+function readFile(path) {
+  return new Promise((resolve, reject) => {
+    fs.readFile(path, 'utf8', (error, data) => {
+      if (error) {
+        reject(error);
+      } else {
+        resolve(data);
+      }
+    });
+  });
+}
+
+function writeFile(path, data) {
+  return new Promise((resolve, reject) => {
+    fs.writeFile(path, JSON.stringify(data), 'utf8', error => {
+      reject(error);
+    });
+  });
+}
+
+function sendResponse(data, response) {
+  response.json(data);
+}
+
+function listToDo(path, response) {
+  readFile(path)
+    .then(data => sendResponse(data, response))
+    .catch(error => response.status(404).send({ error: 'there is an error' }););
+}
+
+async function addToDo(todoItem, path, response) {
+  try {
+    await readFile(path).then(data => {
+      const todoList = JSON.parse(data);
+      const newTodo = {
+        id: uuid(),
+        name: todoItem.item,
+        done: false,
+      };
+      todoList.push(newTodo);
+      writeFile(path, todoList);
+    });
+    listToDo(path, response);
+  } catch {
+    response.status(404).send({ error: 'there is an error' });
+  }
+}
+
+function readToDo(number, path, response) {
+  readFile(path)
+    .then(data => {
+      const todoList = JSON.parse(data);
+      const todo = todoList.find(todo => {
+        return todo.id == number;
+      });
+      todo !== undefined
+        ? sendResponse(todo.name, response)
+        : sendResponse('todo not found', response);
+    })
+    .catch(error => {
+      response.status(404).send({ error: 'there is an error' });;
+    });
+}
+
+async function markAsDone(number, path, response) {
+  try {
+    await readFile(path)
+      .then(data => {
+        const todoList = JSON.parse(data);
+        const todo = todoList.find(todo => {
+          return todo.id == number;
+        });
+        todo.done = true;
```

Comment:
Like what you did in `readToDo` function, you need to check if the requested todo exists.